examples/DGmethods/ex_003_acoustic_wave.jl

```diff
@@ -0,0 +1,533 @@
+
+# ### Preliminaries
+# Load in modules needed for solving the problem
+using MPI
+using Logging
+using LinearAlgebra
+using Dates
+using Printf
+using CLIMA
+using CLIMA.Topologies
+using CLIMA.MPIStateArrays
+using CLIMA.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.Vtk
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+
+# Though not required, here we are explicit about which values we read out the
+# `PlanetParameters` and `MoistThermodynamics`
+using CLIMA.PlanetParameters: planet_radius, grav, MSLP
+using CLIMA.MoistThermodynamics: air_temperature, air_pressure, internal_energy,
+                                 soundspeed_air, air_density, gas_constant_air
+
+# Start up MPI if this has not already been done
+MPI.Initialized() || MPI.Init()
+#md nothing # hide
+
+# If `CuArrays` is in the current environment we will use CUDA, otherwise we
+# drop back to the CPU
+@static if haspkg("CuArrays")
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  CuArrays.allowscalar(false)
+  const DeviceArrayType = CuArray
+else
+  const DeviceArrayType = Array
+end
+#md nothing # hide
+
+# Specify whether to enforce hydrostatic balance at PDE level or not
+const PDE_level_hydrostatic_balance = false
+
+# check whether to use default VTK directory or define something else
+VTKDIR = get(ENV, "CLIMA_VTK_DIR", "vtk")
+
+# Here we setup constants to for some of the computational parameters; the
+# underscore is just syntactic sugar to indicate that these are constants.
+
+# These are parameters related to the Euler state. Here we used the conserved
+# variables for the state: perturbation in density, three components of
+# momentum, and perturbation to the total energy.
+const _nstate = 5
+const _dρ, _ρu, _ρv, _ρw, _dρe = 1:_nstate
+const _statenames = ("δρ", "ρu", "ρv", "ρw", "δρe")
+#md nothing # hide
+
+# These will be the auxiliary state which will contain the geopotential,
+# gradient of the geopotential, and reference values for density and total
+# energy
+const _nauxstate = 6
+const _a_ϕ, _a_ϕx, _a_ϕy, _a_ϕz, _a_ρ_ref, _a_ρe_ref = 1:_nauxstate
+const _auxnames = ("ϕ", "ϕx", "ϕy", "ϕz", "ρ_ref", "ρe_ref")
+#md nothing # hide
+
+#------------------------------------------------------------------------------
+
+# ### Definition of the physics
+#md # Now we define a function which given the state and auxiliary state defines
+#md # the physical Euler flux
+function eulerflux!(F, Q, _, aux, t)
+  @inbounds begin
+    ## extract the states
+    dρ, ρu, ρv, ρw, dρe = Q[_dρ], Q[_ρu], Q[_ρv], Q[_ρw], Q[_dρe]
+    ρ_ref, ρe_ref, ϕ = aux[_a_ρ_ref], aux[_a_ρe_ref], aux[_a_ϕ]
+
+    ρ = ρ_ref + dρ
+    ρe = ρe_ref + dρe
+    e = ρe / ρ
+
+    ## compute the velocity
+    u, v, w = ρu / ρ, ρv / ρ, ρw / ρ
+
+    ## internal energy
+    e_int = e - (u^2 + v^2 + w^2)/2 - ϕ
```

Comment:
This is fine. But we should probably add a function to calculate internal energy from total in `MoistThermodynamics`. @charleskawczynski ?

Reply:
Yes, I'll add a wrapper function for MoistThermo.